api/views/assets.py

```diff
@@ -441,3 +442,55 @@ def get(self, request):
         ] = 'attachment; filename="sample_import_file.csv"'
 
         return response
+
+
+class ExportAssetsDetails(APIView):
+    permission_classes = [IsAuthenticated, IsAdminUser]
+    authentication_classes = (FirebaseTokenAuthentication,)
+
+    def get(self, request):
+        assets = models.Asset.objects.filter(asset_location__name=request.user.location.name)
+        serializer = AssetSerializer(assets, many=True)
+        if len(serializer.data) == 0:
+            return Response({"error": "You have no assets"}, status=400)
+        self.create_sheet(serializer.data)
+        return Response({
+            "Success": "Assets details have been exported successfully, Find file in root folder"}, status=200)
+
+    def create_sheet(self, assets_list):
+        asset_types = []
+        for asset in assets_list:
+            if asset.get('asset_type') not in asset_types:
+                asset_types.append(asset.get('asset_type'))
+
+        workbook = xlsxwriter.Workbook('assets.xlsx')
+        bold = workbook.add_format({'bold': True, 'bg_color': 'silver'})
+        for asset_type in asset_types:
+            worksheet = workbook.add_worksheet(asset_type)
+            worksheet.write('A1', 'Make', bold)
```

Comment:
We should use the constants you added here too. You might need to merge this one first though.